components/Typography/examples/TypographyMaterialScalableStylesViewController.m

```diff
@@ -0,0 +1,173 @@
+    [_typography.headline5 mdc_scaledFontAtDefaultSize],
+    [_typography.headline5 mdc_scaledFontForCurrentSizeCategory],
+    [_typography.headline6 mdc_scaledFontAtDefaultSize],
+    [_typography.headline6 mdc_scaledFontForCurrentSizeCategory],
+  ];
+
+  [[NSNotificationCenter defaultCenter] addObserver:self
+                                           selector:@selector(contentSizeCategoryDidChange:)
+                                               name:UIContentSizeCategoryDidChangeNotification
+                                             object:nil];
+}
+
+- (void)contentSizeCategoryDidChange:(NSNotification *)notification {
+  NSString *sizeCategory = notification.userInfo[UIContentSizeCategoryNewValueKey];
+  NSLog(@"New size category : %@", sizeCategory);
+
+  // Update font array to reflect new size category
+  _styleFonts = @[
+    [_typography.body1 mdc_scaledFontAtDefaultSize],
+    [_typography.body1 mdc_scaledFontForCurrentSizeCategory],
+    [_typography.body2 mdc_scaledFontAtDefaultSize],
+    [_typography.body2 mdc_scaledFontForCurrentSizeCategory],
+    [_typography.caption mdc_scaledFontAtDefaultSize],
+    [_typography.caption mdc_scaledFontForCurrentSizeCategory],
+    [_typography.button mdc_scaledFontAtDefaultSize],
+    [_typography.button mdc_scaledFontForCurrentSizeCategory],
+    [_typography.overline mdc_scaledFontAtDefaultSize],
+    [_typography.overline mdc_scaledFontForCurrentSizeCategory],
+    [_typography.subtitle1 mdc_scaledFontAtDefaultSize],
+    [_typography.subtitle1 mdc_scaledFontForCurrentSizeCategory],
+    [_typography.subtitle2 mdc_scaledFontAtDefaultSize],
+    [_typography.subtitle2 mdc_scaledFontForCurrentSizeCategory],
+    [_typography.headline1 mdc_scaledFontAtDefaultSize],
+    [_typography.headline1 mdc_scaledFontForCurrentSizeCategory],
+    [_typography.headline2 mdc_scaledFontAtDefaultSize],
+    [_typography.headline2 mdc_scaledFontForCurrentSizeCategory],
+    [_typography.headline3 mdc_scaledFontAtDefaultSize],
+    [_typography.headline3 mdc_scaledFontForCurrentSizeCategory],
+    [_typography.headline4 mdc_scaledFontAtDefaultSize],
+    [_typography.headline4 mdc_scaledFontForCurrentSizeCategory],
+    [_typography.headline5 mdc_scaledFontAtDefaultSize],
+    [_typography.headline5 mdc_scaledFontForCurrentSizeCategory],
+    [_typography.headline6 mdc_scaledFontAtDefaultSize],
+    [_typography.headline6 mdc_scaledFontForCurrentSizeCategory],
+  ];
+
+  [self.tableView reloadData];
+}
+
+#pragma mark - UITableViewDataSource
+
+- (NSInteger)numberOfSectionsInTableView:(UITableView *)tableView {
+  return _strings.count;
+}
+
+- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
+  return _styleNames.count;
+}
+
+- (UITableViewCell *)tableView:(UITableView *)tableView
+         cellForRowAtIndexPath:(NSIndexPath *)indexPath {
+  UITableViewCell *cell = [self.tableView dequeueReusableCellWithIdentifier:@"cell"];
```

Comment:
Please be sure to [register the correct class in `viewDidLoad`](https://developer.apple.com/documentation/uikit/uitableview/1614878-dequeuereusablecellwithidentifie?language=objc).


> **Important**
You must register a class or nib file using the registerNib:forCellReuseIdentifier: or registerClass:forCellReuseIdentifier: method before calling this method.

Reply:
Done, and removed outdated if (cell).